File: TodoApp.Client/src/interfaces/Responses/TodoItemResponse.ts
import dayjs from "dayjs";
import type { Tier } from "../../commons/enums/Tier";

export default interface TodoItemResponse {
  id?: string,
  title: string;
  description: string;
  dueDate: dayjs.Dayjs;
  isCompleted: boolean;
  priority: Tier;
  completedOn: dayjs.Dayjs;
  todoListId?: string;

  createdOn?: dayjs.Dayjs;
  modifiedOn?: dayjs.Dayjs;
}

File: TodoApp.Server/src/Todo.DTOs/Responses/TodoItemResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Todo.Domain.Enums;

namespace Todo.DTOs.Responses
{
    public class TodoItemResponse
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime DueDate { get; set; }
        public bool IsCompleted { get; set; }
        public Tier Priority { get; set; }
        public DateTime? CompletedOn { get; set; }
        public DateTime? CreatedOn { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public Guid? TodoListId { get; set; }
    }
}
